<?xml version="1.0"?>
<launch>
  <arg name="rj45" default="false"/>
  <arg name="robot_version" default="c01_2018"/>
  
  <!--<param name="robot_description" command="$(find xacro)/xacro.py '$(find sick_tim)/urdf/example.urdf.xacro'" />-->
<!--  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />-->
  <include file="$(find laser)/launch/transform/$(arg robot_version)_transform.launch">
  </include>
  <node ns="laser" name="base_laser" pkg="sick_tim" type="sick_tim551_2050001" respawn="false" output="screen">
  
  <remap from="scan" to="scan_raw"/>

  <!--<param name="min_ang" type="double" value="-1.92"/>
  <param name="max_ang" type="double" value="1.92"/>-->
  <param name="intensity" type="bool" value="true" />
  <param name="frame_id" type="str" value="laser_link" />
  <param name="range_min" type="double" value="0.05" />
  <param name="range_max" type="double" value="25.0" />
  <!--  load max and min ang from yaml file-->
  <rosparam command="load" file="(find laser)/cfg/$(arg robot_version)_laser_config.yaml" />


  <!-- Older versions of the scanner firmware report an incorrect time_increment.
       If you get a warning about time_increment, uncomment this. -->
  <!--<param name="time_increment" type="double" value="0.000061722" />-->

    <!-- default values: -->
    <!--
      <param name="min_ang" type="double" value="-2.35619449019" />
      <param name="max_ang" type="double" value="2.35619449019" />
      <param name="intensity" type="bool" value="True" />
      <param name="skip" type="int" value="0" />
      <param name="frame_id" type="str" value="laser" />
      <param name="time_offset" type="double" value="-0.001" />
      <param name="publish_datagram" type="bool" value="False" />
      <param name="subscribe_datagram" type="bool" value="false" />
      <param name="device_number" type="int" value="0" />
      <param name="time_increment" type="double" value="0.000061722" />
      <param name="range_min" type="double" value="0.05" />
      <param name="range_max" type="double" value="10.0" />
    -->

    <!-- Uncomment this to enable TCP instead of USB connection; 'hostname' is the host name or IP address of the laser scanner
    In cases where a race condition exists and the computer boots up before the TIM is ready, increase 'timelimit.'
         <param name="hostname" type="string" value="192.168.1.43" />
         <param name="port" type="string" value="2112" />
         <param name="timelimit" type="int" value="5" />
    -->
    <!-- <param name="hostname" type="string" value="169.254.220.230" /> -->
    <param name="hostname" type="string" value="169.254.51.222" if="$(arg rj45)"/>
    <param name="port" type="string" value="2112" if="$(arg rj45)"/>
    <param name="timelimit" type="int" value="5" if="$(arg rj45)"/>
  </node>

<!--                                  -->

</launch>
